File: app/src/main/java/com/example/mycaiculatorwork8/MainActivity.java
package com.example.calculator;

import androidx.appcompat.app.AppCompatActivity;
import androidx.annotation.NonNull;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView display;
    private StringBuilder currentInput = new StringBuilder();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        display = findViewById(R.id.display);

        if (savedInstanceState != null) {
            currentInput.append(savedInstanceState.getString("input"));
            display.setText(currentInput.toString());
        }


        setupButtons();
    }

    private void setupButtons() {
        int[] buttonIds = {
                R.id.button1, R.id.button2, R.id.button3,
                R.id.button4, R.id.button5, R.id.button6,
                R.id.button7, R.id.button8, R.id.button9,
                R.id.button0, R.id.button_add, R.id.button_subtract,
                R.id.button_multiply, R.id.button_divide, R.id.button_clear};

        for (int id : buttonIds) {
            Button button = findViewById(id);
            button.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    handleInput(((Button) v).getText().toString());
                }
            });
        }
    }

    private void handleInput(String input) {
        if (input.equals("C")) {
            currentInput.setLength(0); 
        } else {
            currentInput.append(input); // Добавление нажатой цифры или знака.
        }
        display.setText(currentInput.toString()); // Обновление дисплея.
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putString("input", currentInput.toString()); // Сохранение состояния ввода.
    }
}

File: NoteApp/src/main/java/com/example/notesapp/MainActivity.kt
package com.example.notesapp

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    private var recyclerView: RecyclerView? = null
    private var notesAdapter: NotesAdapter? = null
    private var notesList: MutableList<Note>? = null

    protected override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyclerView = findViewById(R.id.recyclerView)
        notesList = ArrayList()

        // Добавим несколько заметок для тестирования
        notesList.add(Note("Заметка 1", "Содержимое заметки 1"))
        notesList.add(Note("Заметка 2", "Содержимое заметки 2"))
        notesList.add(Note("Заметка 3", "Содержимое заметки 3"))

        notesAdapter = NotesAdapter(notesList)
        recyclerView!!.layoutManager = LinearLayoutManager(this)
        recyclerView!!.adapter = notesAdapter
    }
}